ZenPacks/zenoss/OpenStackInfrastructure/modeler/plugins/zenoss/OpenStackInfrastructure.py

```diff
@@ -656,14 +657,14 @@ def process(self, device, results, unused):
             subnets.append(ObjectMap(
                 modname='ZenPacks.zenoss.OpenStackInfrastructure.Subnet',
                 data=dict(
-                    cidr=subnet['cidr'],
-                    dns_nameservers=subnet['dns_nameservers'],
-                    gateway_ip=subnet['gateway_ip'],
-                    id=prepId('subnet-{0}'.format(subnet['id'])),
-                    set_network=prepId('network-{0}'.format(subnet['network_id'])),
-                    set_tenant=prepId('tenant-{0}'.format(subnet['tenant_id'])),
-                    subnetId=subnet['id'],
-                    title=subnet['name'],
+                    cidr=subnet.get('cidr',''),
+                    dns_nameservers=subnet.get('dns_nameservers', []),
+                    gateway_ip=subnet.get('gateway_ip',''),
+                    id=prepId('subnet-{0}'.format(subnet.get('id', ''))),
+                    set_network=prepId('network-{0}'.format(subnet.get('network_id',''))),
```

Comment:
Might want to check what set_network does with a '', vs a None.  I'm not sure which one it will like.